modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IORunLoop.kt

```diff
@@ -432,20 +432,7 @@ internal object IORunLoop {
     }
 
     override fun resumeWith(result: Result<Any?>) {
-      if (canCall) {
-        canCall = false
-        result.fold(
-          { a -> IO.Pure(a) },
-          { e -> IO.RaiseError(e) }
-        ).let { r ->
-          if (shouldTrampoline) {
-            this.value = r
-            Platform.trampoline { trampoline() }
-          } else {
-            signal(r)
-          }
-        }
-      }
+      invoke(result.fold(::Right, ::Left))
```

Comment:
Can we refactor this in another manner?

If we split it in `fun finishPure` and `fun finishError` like we do in `parMap` & `race`? Then we can `fold` in both `resumeWith` and `invoke` and avoid the additional wrapping we introduced now.

Reply:
fine with me, i moved the let part to an inline function. cc @pakoito